lib/merkle/merkle.go

```diff
@@ -0,0 +1,47 @@
+// Package merkle contains a n-ary merkletree implementation
+package merkle
+
+import (
+	"encoding"
+
+	"github.com/dedis/crypto/abstract"
+)
+
+// Merkle contains all information needed to hash the data into a commit-tree
+type Merkle struct {
+	suite abstract.Suite
```

Comment:
Do we really need the suite ? To make it more portable, can't we just take a Hash function as the field ? Since we only use the Hash function anyway + we do not instantiate anything with the suite.